modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -690,18 +690,35 @@ private String getNameUsingEnumPropertyNaming(String name) {
             case original:
                 return name;
             case camelCase:
-                return camelize(name, true);
+                return camelize(prepareEnumString(name), true);
             case PascalCase:
-                return camelize(name);
+                return camelize(prepareEnumString(name));
             case snake_case:
                 return underscore(name);
             case UPPERCASE:
-                return name.toUpperCase(Locale.ROOT);
+                return prepareEnumString(name).toUpperCase(Locale.ROOT);
             default:
                 throw new IllegalArgumentException("Unsupported enum property naming: '" + name);
         }
     }
 
+    private String prepareEnumString(String name) {
```

Comment:
please add some docs to explain what this function does, and include an example

I am just asking myself: why can't we use the `underscore` method?
https://github.com/OpenAPITools/openapi-generator/blob/eb62e21d8f0cf77e415dc2864c011c59290444bd/modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/StringUtils.java#L59

Reply:
Yes, I can use the underscore method. I deleted prepareEnumString method. Thanks.